squire_lib/src/ffi_rounds.rs

```diff
@@ -0,0 +1,160 @@
+        match self.get_tourn_round(tid) {
+            Some(r) => {
+                return r.table_number as i64;
+            }
+            None => {
+                return -1;
+            }
+        }
+    }
+
+    /// Gets the status for a round
+    /// Dead on error
+    #[no_mangle]
+    pub extern "C" fn rid_status(self, tid: TournamentId) -> RoundStatus {
+        match self.get_tourn_round(tid) {
+            Some(r) => {
+                return r.status;
+            }
+            None => {
+                return RoundStatus::Dead;
+            }
+        }
+    }
+
+    /// Returns the amount of time left in a round
+    /// Retrusn -1 on error
+    #[no_mangle]
+    pub extern "C" fn rid_time_left(self, tid: TournamentId) -> i64 {
+        match self.get_tourn_round(tid) {
+            Some(r) => {
+                return r.time_left().as_secs() as i64;
+            }
+            None => {
+                return -1;
+            }
+        }
+    }
+
+    /// Returns the total duration in a round
+    /// Retrusn -1 on error
+    #[no_mangle]
+    pub extern "C" fn rid_duration(self, tid: TournamentId) -> i64 {
+        match self.get_tourn_round(tid) {
+            Some(r) => {
+                return r.length.as_secs() as i64;
+            }
+            None => {
+                return -1;
+            }
+        }
+    }
+
+    /// Gets the players that are in a round
+    /// Returns NULL on error
+    #[no_mangle]
+    pub extern "C" fn rid_players(self, tid: TournamentId) -> *const PlayerId {
+        match self.get_tourn_round(tid) {
+            Some(r) => {
+                let players: HashSet<PlayerId> = r.players;
+                let len: usize = (players.len() + 1) * std::mem::size_of::<PlayerId>();
+
+                unsafe {
+                    let ptr = System
+                        .allocate(Layout::from_size_align(len, 1).unwrap())
+                        .unwrap()
+                        .as_mut_ptr() as *mut PlayerId;
+                    let slice = &mut *(ptr::slice_from_raw_parts(ptr, len) as *mut [PlayerId]);
+
+                    let mut i: usize = 0;
+                    for p in players {
+                        slice[i] = p;
+                        i += 1;
+                    }
+                    slice[i] = Uuid::default().into();
+                    return ptr;
+                }
+            }
+            None => {
+                return std::ptr::null();
+            }
+        }
+    }
+
+    /// Gets the players that are in a round who have confirmed
+    /// Returns NULL on error
+    #[no_mangle]
+    pub extern "C" fn rid_confirmed_players(self, tid: TournamentId) -> *const PlayerId {
+        match self.get_tourn_round(tid) {
+            Some(r) => {
+                let players: HashSet<PlayerId> = r.confirmations;
+                let len: usize = (players.len() + 1) * std::mem::size_of::<PlayerId>();
+
+                unsafe {
```

Comment:
Similar change here.

Reply:
Fixed in e726d5c5f3b0849328045ea3854859a3578bae01